test/extensions/clusters/redis/redis_cluster_test.cc

```diff
@@ -433,10 +431,10 @@ class RedisClusterTest : public testing::Test,
     discovery_session.registerDiscoveryAddress(
         TestUtility::makeDnsResponse(std::list<std::string>({"127.0.0.1", "127.0.0.2"})), 22120);
     expectRedisResolve(true);
-    discovery_session.startResolve();
+    discovery_session.startResolveRedis();
 
-    // 2nd startResolve() call will be a no-opt until the first startResolve is done.
-    discovery_session.startResolve();
+    // 2nd startResolveRedis() call will be a no-opt until the first startResolve is done.
+    discovery_session.startResolveRedis();
```

Comment:
What is this test testing? I'm confused why we are reaching into this class to test it like this?

Reply:
As the comment said, this is when we have 2nd startResolveRedis calls before the first one finishes, it'll be a no-opt.  This is needed to hit 100% test coverage you asked for.

Reply:
But if this can't happen, why do we need a test? Just switch whatever logic you have to an assert?

Reply:
This is to cover multiple DNS entries in the configuration.  That's a valid case.

Reply:
Then why can't you have a case that covers this using mocks, etc? I don't understand why you are reaching into the class.

Reply:
This behavior is on the RedisCluster::RedisDiscoverySession class and the unit test is done by calling the function on the RedisDiscoverySession class, and we're using a mock client in the expectRedisResolve() function.

Reply:
Sorry let me be more specific. Why do you need to reach into the private innards of the class to cover this case? I believe you can remove the friend status for the test and actually test this via the cluster itself which would be a substantially better test?

/wait

Reply:
There're a number of test cases that'd be simpler if we test the private class directly, so we'll still need the friend class.  In this particular case, you're right in that it's possible to cover the case without testing the private class directly.  The test would have to mock the dns resolvers and a bit more complicated and the destructor expectation would be less clear than if we explicitly test the private class directly.  

If you want me to change the test, I can do it with a different PR.  This seems to be an unrelated issue to this PR.

Reply:
That's fine, please do a follow up to improve the tests before doing further feature work. Thank you!